feature: add ha overlay for seaweedfs

pschoen-itsc opened the pull request:
**Description of your changes:**


**Checklist:**
- [x] You have [signed off your commits](https://www.kubeflow.org/docs/about/contributing/#sign-off-your-commits)
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 
<!--
   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.
-->

pschoen-itsc committed:
Add ha overlay for seaweedfs

Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

# Conflicts:
#	manifests/kustomize/third-party/seaweedfs/base/seaweedfs/kustomization.yaml
#	manifests/kustomize/third-party/seaweedfs/base/seaweedfs/seaweedfs-create-admin-user-job.yaml

juliusvonkohout commented:
And please test directly with 4.0.0 as in https://github.com/kubeflow/pipelines/pull/12406

juliusvonkohout commented:
https://artifacthub.io/packages/helm/bitnami/seaweedfs/4.0.0 and even more https://github.com/seaweedfs/seaweedfs/tree/master/k8s/charts/seaweedfs might be a good comparision

pschoen-itsc committed:
Add image tag to kustomize

Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

pschoen-itsc committed:
Fix seaweedfs single-pod path

Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

pschoen-itsc committed:
Use preferred affinity instead of required

Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

hbelmiro committed:
fix(backend): Parse JSON and decode base64 in artifact client (#12467)

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

pschoen-itsc committed:
Rebase onto master

Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

pschoen-itsc committed:
Fix master url

Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

HumairAK committed:
chore: update release notes and scripts (#12470)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

juliusvonkohout committed:
Add fsGroup and seaweedfs (#12474)

As we also have it in https://github.com/kubeflow/manifests/blob/c0c90500c6506d1acba16fbba5f7d3bff9d90d31/applications/pipeline/upstream/third-party/minio/base/minio-deployment.yaml#L19

Signed-off-by: Julius von Kohout <45896133+juliusvonkohout@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

nsingla committed:
ci: fixing and reenabling upgrade tests (#12475)

Signed-off-by: Nelesh Singla <117123879+nsingla@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

nsingla committed:
ci: adding a retry logic to create cluster step in case it fails (#12476)

Signed-off-by: Nelesh Singla <117123879+nsingla@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

HumairAK committed:
chore: remove reusable publishing code, pypi does not allow it (#12482)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

HumairAK committed:
Add release 2.15 changes to master (#12486)

* chore(release): bumped version to 2.15.0 (#12472)

* chore(release): bumped version to 2.15.0

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* bump python package versions to 2.15.0

also ensure kfp packages are being installed from source code from CI.

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

---------

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* chore: update kfp package conf versions and docs (#12485)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* revert tools images back to master from 2.15 change

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

---------

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

alyssacgoins committed:
configure ml-pipeline service address within driver.go (#12481)

Signed-off-by: alyssacgoins <agoins@redhat.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

droctothorpe committed:
fix(sdk): align sdk versions in reqs (#12489)

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

nsingla committed:
ci: Git artifact check (#12477)

Signed-off-by: Nelesh Singla <117123879+nsingla@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

akagami-harsh committed:
chore: remove duplicate default-allow-same-namespace network policy (#12492)

* remove default-allow-same-namespace network policy

Signed-off-by: Harshvir Potpose <hpotpose62@gmail.com>

* add network policy in base test manifests

Signed-off-by: Harshvir Potpose <hpotpose62@gmail.com>

---------

Signed-off-by: Harshvir Potpose <hpotpose62@gmail.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

HumairAK committed:
chore: add commit step to kfp k8s branch creation script changes (#12500)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

nsingla committed:
ci: add path references verification in the github workflows to CI/CD sec… (#12502)

Signed-off-by: Nelesh Singla <117123879+nsingla@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

uzi0espil committed:
fix(sdk): fix set_container_image within ParallelFor. Fixes #12413 (#12414)

* sdk: fix set_container_image within ParallelFor

Signed-off-by: Ousama Esbel <ousama.esbel@gmail.com>

* style: reformat test to adhere to yapf

Signed-off-by: Ousama Esbel <ousama.esbel@gmail.com>

---------

Signed-off-by: Ousama Esbel <ousama.esbel@gmail.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

nsingla committed:
ci: adding concurrency behavior to cancel in progress workflows if a new … (#12504)

Signed-off-by: Nelesh Singla <117123879+nsingla@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

nsingla committed:
ci: adding a github action to generate test summary with html report link attached (#12478)

Signed-off-by: Nelesh Singla <117123879+nsingla@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

mprahl committed:
fix(sdk): Set custom_path in executor output only when used (#12514)

Prior to this commit, the executor output always set a custom_path field
regardless of if it was set. This caused KFP API server versions prior
to 2.15 to fail in the launcher with:
"output_metadata.json": proto: (line 1:237): unknown field "custom_path"

Signed-off-by: mprahl <mprahl@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

mprahl committed:
chore: Sync KFP SDK 2.15.2 in master (#12518)

Signed-off-by: mprahl <mprahl@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

dependabot[bot] committed:
chore(deps): bump urllib3 from 2.5.0 to 2.6.0 in /backend/metadata_writer (#12531)

Bumps [urllib3](https://github.com/urllib3/urllib3) from 2.5.0 to 2.6.0.
- [Release notes](https://github.com/urllib3/urllib3/releases)
- [Changelog](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst)
- [Commits](https://github.com/urllib3/urllib3/compare/2.5.0...2.6.0)

---
updated-dependencies:
- dependency-name: urllib3
  dependency-version: 2.6.0
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

juliusvonkohout committed:
Enhance security context in cache-deployer deployment (#12538)

* Enhance security context in cache-deployer deployment

Added security context to the cache-deployer deployment.

Signed-off-by: Julius von Kohout <45896133+juliusvonkohout@users.noreply.github.com>

* Enhance cache-deployer deployment with labels

Add metadata labels to the cache-deployer template.

Signed-off-by: Julius von Kohout <45896133+juliusvonkohout@users.noreply.github.com>

* Remove app label from cache-deployer metadata

Removed metadata labels from cache-deployer deployment.

Signed-off-by: Julius von Kohout <45896133+juliusvonkohout@users.noreply.github.com>

---------

Signed-off-by: Julius von Kohout <45896133+juliusvonkohout@users.noreply.github.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

sduvvuri1603 committed:
Cleanup kind workflows and Harden release downloads(incl. artifact paging) (#12549)

* chore: add cleanup jobs for kind workflows

Signed-off-by: sduvvuri1603 <sduvvuri@redhat.com>

* chore: cleanup kind workflows and harden artifacts

Signed-off-by: sduvvuri1603 <sduvvuri@redhat.com>

---------

Signed-off-by: sduvvuri1603 <sduvvuri@redhat.com>
Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

pschoen-itsc committed:
Set artifact proxy to false for test

Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

pschoen-itsc committed:
Merge branch 'master' of github.com:kubeflow/pipelines into seaweedfs-ha-overlay

Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

pschoen-itsc committed:
Fix deploy kfp script and restructure test manifests

Signed-off-by: Patrick Schönthaler <patrick.schoenthaler@itsc.de>

pschoen-itsc committed:
Merge branch 'master' into seaweedfs-ha-overlay
